### PR TITLE
Send update from the scratch

### PR DESCRIPTION
**Pull request by mrutid**

### CODE REVIEW

**Comment by AlvaroVega** [comment]
It would be nice create two PRs for iotagent-json and iotagent-ul to use this branch instead master of iotagetn-node-lib to see how impacts to iotagents.
Changes to remove autocast are still not in master,but already in this PR. Autocast impacts in iotagent-ul and still is not fixed there.

File: lib/commonConfig.js
Comment (fgalan) [inline lib/commonConfig.js:457]: This change seems similar to the ones in PR https://github.com/telefonicaid/iotagent-node-lib/pull/1501. Is this PR based on PR #1501? Or it is just it has touched the same code but without direct relationship?

Whichever the case, it should include proper CHANGES_NEXT_RELEASE entry:

```
- Remove: autocast (including env var IOTA_AUTOCAST) (#1498)
```

Comment (fgalan) [inline lib/commonConfig.js:457]: This change seems similar to the ones in PR https://github.com/telefonicaid/iotagent-node-lib/pull/1501. Is this PR based on PR #1501? Or it is just it has touched the same code but without direct relationship?

Whichever the case, it should include proper CHANGES_NEXT_RELEASE entry:

```
- Remove: autocast (including env var IOTA_AUTOCAST) (#1498)
```

File: package.json
Comment (fgalan) [inline package.json:38]: Should this be included at the end? Or it was just for debugging?

In the former case, I'd suggest to use `test:jexl` (style)

File: test/unit/ngsiv2/plugins/alias-plugin_test.js
Comment (fgalan) [inline test/unit/ngsiv2/plugins/alias-plugin_test.js:1]: We have some *test.js files under ngsiv2/plugins directory, but I understand plugin architecture was removed with the refactor (not fully sure, I may be wrong). In that case, this directory hierarchy should be simplified (although maybe not in this PR).

File: test/unit/ngsiv2/plugins/compress-timestamp-plugin_test.js
Comment (fgalan) [inline test/unit/ngsiv2/plugins/compress-timestamp-plugin_test.js:121]: xdescribe() should be solved before this PR gets merged.

Comment (fgalan) [reply test/unit/ngsiv2/plugins/compress-timestamp-plugin_test.js:121]: 3 cases in this file.

File: test/unit/ngsiv2/plugins/multientity-plugin_test.js
Comment (fgalan) [inline test/unit/ngsiv2/plugins/multientity-plugin_test.js:1506]: xdescribe() should be solved before this PR gets merged.

File: test/unit/ngsiv2/plugins/timestamp-processing-plugin_test.js
Comment (fgalan) [inline test/unit/ngsiv2/plugins/timestamp-processing-plugin_test.js:81]: xdescribe() should be solved before this PR gets merged.

File: doc/admin.md
Comment (fgalan) [inline doc/admin.md:22]: As a general comment in this PR, being a mayor refactor of the code, I think it would be a good idea to mention in the CHANGES_NEXT_RELEASE file. For instance:

```
- Large refactor of IOTA Lib code to make it simpler
```

or something similar.

Probably this deserves a new mayor version (4.0.0 version).

**Review by fgalan — CHANGES_REQUESTED** [review]

File: lib/commonConfig.js
Comment (mrutid) [reply lib/commonConfig.js:457]: I had synced with master and with removeautocast2 branch during refactor to take advantage of Alvaro's parallel working

File: package.json
Comment (mrutid) [reply package.json:38]: debugging I will removeit

File: test/unit/ngsiv2/plugins/alias-plugin_test.js
Comment (mrutid) [reply test/unit/ngsiv2/plugins/alias-plugin_test.js:1]: the plugins have been removed but no the functionality. Maybe the dir shpuld be remamed (minor for me)

File: doc/admin.md
Comment (fgalan) [reply doc/admin.md:22]: Timecompression functionality has been also removed? In that case, CNR entry like this could be added:

```
- Remove: time compression support
```

**Comment by fgalan** [comment]
Not for version 3.5.0

File: test/unit/ngsiv2/plugins/compress-timestamp-plugin_test.js
Comment (mrutid) [reply test/unit/ngsiv2/plugins/compress-timestamp-plugin_test.js]: removed. Deprecated functionaliy

File: package.json
Comment (mrutid) [reply package.json]: solved

File: test/unit/ngsiv2/plugins/alias-plugin_test.js
Comment (fgalan) [reply test/unit/ngsiv2/plugins/alias-plugin_test.js:1]: NTC

**Review by fgalan — APPROVED** [review]
LGTM

Let's have LGTM first in PR https://github.com/telefonicaid/iotagent-json/pull/765 before merging.

**Comment by AlvaroVega** [comment]
> LGTM
> 
> Let's have LGTM first in PR [telefonicaid/iotagent-json#765](https://github.com/telefonicaid/iotagent-json/pull/765) before merging.

Already iotagent PR are ready to merge:
https://github.com/telefonicaid/iotagent-ul/pull/645
https://github.com/telefonicaid/iotagent-json/pull/765

**Comment by fgalan** [comment]
> > LGTM
> > Let's have LGTM first in PR [telefonicaid/iotagent-json#765](https://github.com/telefonicaid/iotagent-json/pull/765) before merging.
> 
> Already iotagent PR are ready to merge: [telefonicaid/iotagent-ul#645](https://github.com/telefonicaid/iotagent-ul/pull/645) [telefonicaid/iotagent-json#765](https://github.com/telefonicaid/iotagent-json/pull/765)

Also this one as regression for the IOTA Manager: PR https://github.com/telefonicaid/iotagent-manager/pull/322 (to be removed after successful regression)

**Pull request merged**
